### ISSUE TITLE
Scripting variables (request, httpRegion.request) not completely initialized

### ISSUE THREAD

**Issue by wusawo** [AnWeber/httpyac#359]
Hi @AnWeber 

the `body` and/or `rawBody` on the httpyac variable `request` and `httpRegion.request` are not always set.

```
###
# @name postHttpbin2

{{
    console.info(`BEFORE request - current request rawBody: ${request.rawBody}`); 
    console.info(`BEFORE request - current request body: ${request.body}`); 

    httpRegion1 = httpFile.httpRegions.find(r => r.metaData.name === "postHttpbin1");
    console.info(`BEFORE request - httpRegion1 request rawBody: ${httpRegion1.request.rawBody}`); 
    console.info(`BEFORE request - httpRegion1 request body: ${httpRegion1.request.body}`); 

}}
POST https://httpbin.org/post?source=postHttpbin2 HTTP/1.1

{
    "foo": "postHttpbin2"
}

{{
    console.info(`AFTER request - current request rawBody: ${request.rawBody}`); 
    console.info(`AFTER request - current request body: ${request.body}`); 

    httpRegion1 = httpFile.httpRegions.find(r => r.metaData.name === "postHttpbin1");
    console.info(`AFTER request - httpRegion1 request rawBody: ${httpRegion1.request.rawBody}`); 
    console.info(`AFTER request - httpRegion1 request body: ${httpRegion1.request.body}`); 
}}


###
# @name postHttpbin1
POST https://httpbin.org/post HTTP/1.1

{
    "foo": "postHttpbin1"
}
```

Results in the following output:
```
INFO: BEFORE request - current request rawBody: undefined
INFO: BEFORE request - current request body: {
    "foo": "postHttpbin2"
}
INFO: BEFORE request - httpRegion1 request rawBody: undefined
INFO: BEFORE request - httpRegion1 request body: undefined
INFO: AFTER request - current request rawBody: undefined
INFO: AFTER request - current request body: {
    "foo": "postHttpbin2"
}
INFO: AFTER request - httpRegion1 request rawBody: undefined
INFO: AFTER request - httpRegion1 request body: undefined
```

Thank you

**Comment by AnWeber** [comment]
@wusawo `request.rawBody` is not used anymore. I removed property in interface.

To explain the difference between `request` and `httpRegion.request` some more explanation is needed. 
1. `request` is a [clone of `httpRegion.request` and is created at start of executing a region](https://github.com/AnWeber/httpyac/blob/2be987e9b33688f901f95bb48293cf54dbceaff0/src/plugins/core/execute/createRequestInterceptor.ts#L14). For this reason you should not change `httpRegion.request` directly, but only `request`.
2. After request clone the body [is lazy loaded to minimize memory consumption](https://github.com/AnWeber/httpyac/blob/2be987e9b33688f901f95bb48293cf54dbceaff0/src/plugins/core/parseEnd/createRequestBodyInterceptor.ts#L13). This means on `httpRegion.request` the body property is never initialized. At this stage the body can have [different data types](https://github.com/AnWeber/httpyac/blob/f4f3af8367477de6ea6bbe06d2b9277504a9e15d/src/models/httpRequest.ts#L19). 
3. your script is executed
4. before executing the real request the body is normalized to data type [Buffer or string](https://github.com/AnWeber/httpyac/blob/f4f3af8367477de6ea6bbe06d2b9277504a9e15d/src/plugins/core/request/transformRequestBodyAction.ts#L5).
5. I did not think about the fact that after execution of the http request the variable `request` can still be accessed. This remains unchanged after `transform`, but this would be another way to free up memory. I would advise against using this and instead define a separate variable for possible accesses

In summary: `httpRegin.request.body` is never assigned. Don't touch `httpRegion.request`. `request.body` is lazy loaded and should be managed in custom variable, if same value and type is needed after request.

**Comment by wusawo** [comment]
I need to access the request body of one httpRegion from another one (see my sample above). Is there a chance to force loading the body of that httpRegion? Any other ideas?

**Comment by AnWeber** [comment]
If the other request is already executed, you could save the request body in a variable. Or just use a script to init body in variable.

```
{{
    exports.body= {
    "foo": "postHttpbin2"
    };
}}
###
# @name postHttpbin2

POST https://httpbin.org/post?source=postHttpbin2 HTTP/1.1

{{body}}


###
# @name postHttpbin1
POST https://httpbin.org/post HTTP/1.1

{{body}}
```

**Comment by wusawo** [comment]
That's a pity but not so bad. Actually I am using httpyac's domain specific language for http requests to express test data, because in one of my scenarios the test data is a http request definition. This test data request is never executed but I use its method, uri and body in an actual executed request as variables by referring to the particular httpRegion `httpFile.httpRegions.find(r => r.metaData.name === "postHttpbin1")`  😉 
Of course to workaround this issue I can also define exports for testDataRequest.Method, testDataRequest.Uri and testDataRequestBody

**Comment by AnWeber** [comment]
With v5.10.0 I now already append the body to the httpRegion during parsing. This can be either a string or an array of string and functions. Your use case should work with this.  Also please note that the variables in this body have not been replaced yet.

**Comment by wusawo** [comment]
Yes, that works perfectly for my scenario. Thank you.
